product-service/handlers/catalogBatchProcess.ts

```diff
@@ -0,0 +1,52 @@
+import { SNS } from 'aws-sdk'
+import { createProduct, getProductsByTitle, isValidProduct } from './services/productsService'
+
+const sendProductCreatedNotification = (sns, product) => {
+    console.log('Sending product created notification to SNS Topic')
+    sns.publish({
+        Subject: `NodeJS in AWS training: Product ${product.title} was created`,
+        Message: `NodeJS in AWS training: Product ${JSON.stringify(product)} was created in DB`,
+        MessageAttributes: {
+            productCount: {
+                DataType: 'Number',
+                StringValue: product.count.toString(),
+            },
+        },
+        TopicArn: process.env.SNS_ARN 
+    }, (error, data) => {        
+        if (error) {
+            console.log(`Couldn't send product to SNS, error: ${error}`)
+        } else {
+            console.log(`Send message to SNS with product created: ${JSON.stringify(data)}`)
+        }
+    })
+}
+
+export default async event => {
+    console.log("catalogBatchProcess Lambda started execution");
+    console.log("EVENT\n" + JSON.stringify(event, null, 2));
+
+    const sns = new SNS({ region: 'eu-west-1'})
+    return Promise.all(event.Records.map(async ({ body }) => {
```

Comment:
It will be more correct to use Promise.allSettled instead of Promise.all. 
In the case of the error inside the async function Promise.all will immediately become rejected. It will cause the AWS to think that handler execution was stopped, but the event loop may still have events. At least it can result in messing logs inside CloudWatch.